cpp/daal/src/algorithms/dtrees/forest/classification/df_classification_predict_dense_default_batch_impl.i

```diff
@@ -493,25 +567,27 @@ DAAL_FORCEINLINE void PredictClassificationTask<float, avx512>::predictByTree(co
         }
         const double * probas = _model->getProbas(iTree);
 
-        fillResults<float, avx512>(_nClasses, _votingMethod, _DEFAULT_BLOCK_SIZE, probas, left_son, idx, resPtr);
+        fillResults<float, avx512>(_nClasses, _votingMethod, _blockSize, probas, left_son, idx, resPtr);
+        return Status();
     }
     else
     {
-        predictByTreeCommon(x, sizeOfBlock, nCols, feat_idx, left_son, split_point, resPtr, iTree);
+        return predictByTreeCommon(x, sizeOfBlock, nCols, feat_idx, left_son, split_point, resPtr, iTree);
     }
 }
 
 template <>
-DAAL_FORCEINLINE void PredictClassificationTask<double, avx512>::predictByTree(const double * const x, const size_t sizeOfBlock, const size_t nCols,
-                                                                               const featureIndexType * const feat_idx,
-                                                                               const leftOrClassType * const left_son,
-                                                                               const double * const split_point, double * const resPtr,
-                                                                               const size_t iTree)
+DAAL_FORCEINLINE Status PredictClassificationTask<double, avx512>::predictByTree(const double * const x, const size_t sizeOfBlock, const size_t nCols,
+                                                                                 const featureIndexType * const feat_idx,
+                                                                                 const leftOrClassType * const left_son,
+                                                                                 const double * const split_point, double * const resPtr,
+                                                                                 const size_t iTree)
 {
-    if (sizeOfBlock == _DEFAULT_BLOCK_SIZE)
+    if (sizeOfBlock == _blockSize)
```

Comment:
is it necessary to add some checks for _blocksSize somewhere in DAAL part?

Reply:
The checks are implemented here:
https://github.com/oneapi-src/oneDAL/pull/2692/files/596158e2645e872ad4232142734dad0eef582c1c#diff-f7c3689a30067e4f500431476e1897d6949edfa783c36c86a86805d4267d6cd3R282